### PR TITLE
Add distributed sparse matrix-vector dot

### PR DESCRIPTION
**Pull request by wongalvis14**
Add distributed sparse matrix-vector dot

### CODE REVIEW

File: modules/packages/LinearAlgebra.chpl [hunk 1]
```diff
@@ -1678,6 +1678,49 @@ module Sparse {
...
+  
+  /* Sparse Matrix-vector multiplication with distributed support */
+  proc _sparsematvecMult(const ref A: [?Adom] ?eltType, const ref X: [?Xdom] eltType) where isSparseArr(A) {
+    var Y: [Xdom] eltType;
+    var locks$: [LocaleSpace] sync bool;
+    coforall l in Locales do on l {
+      const localDomain = A.localSubdomain();
+      var (low_i, low_j) = localDomain.low;
+      var (high_i, high_j) = localDomain.high;
+      var rowResults: [low_i..high_i] eltType = 0;
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl]: Now, I think we can rename this variable to localResults, or something.

File: modules/packages/LinearAlgebra.chpl [hunk 2]
```diff
@@ -1678,6 +1678,49 @@ module Sparse {
...
+
+      forall (i,j) in localDomain {
+          rowResults[i] += A[i,j] * X[j];
+      }
+      
+      // Writing result
+      // This part assumes 0-based indexing of Adom
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl]: Is this still true? Or is it something forgotten from an earlier version? I don't see anything specific that is based on this assumption but I may be missing something.

File: modules/packages/LinearAlgebra.chpl [hunk 3]
```diff
@@ -1678,6 +1678,49 @@ module Sparse {
...
+      const numRowsPerSynch = localDomain.dim(1).size / A.domain.dist.targetLocales().shape(2);
+      const targetLoc = A.domain.dist.targetLocales();
+      var numLocsInRow = targetLoc.shape(2);
+      var rowid, colid: int;
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl]: we can camelCase these: rowId, colId

File: modules/packages/LinearAlgebra.chpl [hunk 4]
```diff
@@ -1678,6 +1678,49 @@ module Sparse {
...
+      for (ind, Loc) in zip(targetLoc.domain, targetLoc) {
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl]: Loc -> loc

Comment (e-kayrakli) [reply modules/packages/LinearAlgebra.chpl]: I think this loop deserves a small comment. This loop is trying to find where the locale is in the targetLocales array in a rather naive way. There must be a support for that in the distributions to get that without having to do this manually.

File: modules/packages/LinearAlgebra.chpl (hunk 2 above)
Comment (wongalvis14) [reply modules/packages/LinearAlgebra.chpl]: Maybe it can work with arbitrary indexing after some small edits, but I am afraid my math will be incorrect

**Comment by e-kayrakli** [comment]
This PR still needs a test. It'll help me review as well.

**Comment by e-kayrakli** [comment]
@wongalvis14 -- I am closing this PR because it has been very active. Please feel free to discuss here about any questions you may have and reopen if you make some progress on it. Thanks!

**Pull request closed without merging**